src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<String> readFile(String filePath) {
+        if (filePath == null || !filePath.endsWith("csv")) {
```

Comment:
Consider extracting this checks to a separate method e.g
`private void validateFilePath(String filePath) {
        Objects.requireNonNull(filePath, "File path cannot be null");
        
        if (!filePath.endsWith(".csv")) {
            throw new IllegalArgumentException("File must be a CSV file");
        }

        if (!Files.exists(Path.of(filePath))) {
            throw new IllegalArgumentException("File does not exist: " + filePath);
        }
    }`